unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,225 @@
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "unregister": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_unregister(message)
+
+    def test_on_unregister_throws_exception(self, redis_client_auto_register):
+        """Checks if response is not success should throw exception """
+        with pytest.raises(RedisAPIException,
+                           match=f"Failed to unregister from market {AREA_ID}."
+                                 "Deactivating connection."):
+            data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID,
+                    "response": "unsuccessful"}
+            message = {"data": json.dumps(data)}
+            redis_client_auto_register._blocking_command_responses = {
+                "unregister": {"transaction_id": TRANSACTION_ID}}
+            redis_client_auto_register._on_unregister(message)
+
+    def test_select_aggregator(self, redis_client_auto_register):
+        """Checks if redis_db.publish is called with correct arguments to select aggregator."""
+        aggregator_uuid = str(uuid.uuid4())
+        self.redis_client = redis_client_auto_register
+        self.redis_client.redis_db.publish = MagicMock()
+        self.redis_client.area_uuid = AREA_ID
+        self.redis_client.select_aggregator(aggregator_uuid, is_blocking=False)
+        transaction_id = self.redis_client._transaction_id_buffer[0]
+        data = {"aggregator_uuid": aggregator_uuid,
+                "device_uuid": AREA_ID,
+                "type": "SELECT",
+                "transaction_id": transaction_id}
+        self.redis_client.redis_db.publish.assert_called_once_with("aggregator", json.dumps(data))
+
+    def test_select_aggregator_area_uuid_none_throws_exception(self,
+                                                               redis_client_auto_register):
+        """Checks when the device is not registered to the aggregator throws exception."""
+        aggregator_uuid = str(uuid.uuid4())
+        with pytest.raises(RedisAPIException,
+                           match="The device/market has not ben registered yet,"
+                                 " can not select an aggregator"):
+            redis_client_auto_register.select_aggregator(
+                aggregator_uuid=aggregator_uuid, is_blocking=True)
+
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_select_aggregator_with_area_uuid_throws_exception(self,
+                                                               mock_wait_until_timeout_blocking,
+                                                               redis_client_auto_register):
+        """Checks if registered user has a area uuid throws exception."""
+        aggregator_uuid = str(uuid.uuid4())
+        with pytest.raises(RedisAPIException,
+                           match="API has timed out."):
```

Comment:
This exception is only raised because there is no d3a running.
So I would rename the test to `select_aggregator_throws_exception_if_no_d3a_is_running`
And the docstring accordingly.

Reply:
Sure, I will adapt to these changes.